TestAssignmentDoItSoftware/TestAssignmentDoItSoftware/TASingleton/TASingleton.m

```diff
@@ -0,0 +1,596 @@
+//
+//  TASingleton.m
+//  TestAssignmentDoItSoftware
+//
+//  Created by Vladimir Samoylenko on 2/15/19.
+//  Copyright © 2019 TheRedQueen. All rights reserved.
+//
+
+#import "TASingleton.h"
+
+@implementation TASingleton
```

Comment:
As was noted in your original feedback – and something I wholeheartedly agree with – this class should be split into individual services for networking, storage, and data flow concerns.